cpp/src/arrow/flight/sql/odbc/README

```diff
@@ -28,16 +47,16 @@ If the generation is successful, you will find `Apache Arrow Flight SQL ODBC-<ve
 
 ## Steps to Enable Logging
 Arrow Flight SQL ODBC driver uses Arrow's internal logging framework. By default, the log messages are printed to the terminal.
-1. Set environment variable `ARROW_ODBC_LOG_LEVEL` to any of the following valid value.
-
-| LogLevel | Description |
-|----------|-------------|
-| -2 | TRACE |
-| -1 | DEBUG |
-| 0 | INFO |
-| 1 | WARNING |
-| 2 | ERROR |
-| 3 | FATAL |
-| Empty String or not set | No log messages displayed.|
+1. Set environment variable `ARROW_ODBC_LOG_LEVEL` to any of the following valid value to enable logging. If `ARROW_ODBC_LOG_LEVEL` is set to a non-empty string that does not match the following value, `DEBUG` level is used by default. 
+
+The characters are case-insensitive.
+- TRACE
+- DEBUG 
+- INFO
+- WARNING
+- ERROR 
+- FATAL
```

Comment:
How come we are getting rid of the numeric `LogLevel` values?

Reply:
It is because existing Arrow code uses characters for log level instead of numbers. I took the suggestion from community comment: https://github.com/apache/arrow/pull/47645#discussion_r2378046548

Current Arrow code example
https://github.com/bit-quill/arrow/blob/235032ad245030c6364a9c8ec02066c0aa0bb18d/cpp/src/arrow/filesystem/s3fs.cc#L3567-L3585